### ISSUE TITLE
NA in shape index

### ISSUE THREAD

**Issue by mhesselbarth** [r-spatialecology/landscapemetrics#136]
It can happen that NAs are introduced on patch level for the shape index. Currently, this results also in NA on class- and landscape level because we do not remove the NAs when calculating the mean or standard deviation. 

Should we just remove NAs for calculating the mean, sd, etc. and return a warning? I don't know how FRAGSTATS handles that case but will check.

**Comment by mhesselbarth** [comment]
f5a6a01e9b0c2e212de7f87fb847ca8c922c5502

**Comment by mhesselbarth** [comment]
Okay...FRAGSTATS doesn't introduce any NAs at all. I think there seems to be some problem with the minimum perimeter based on the if-else ([see here](http://www.umass.edu/landeco/research/fragstats/documents/Metrics/Shape%20Metrics/Metrics/P8%20-%20SHAPE.htm)). 

One quick fix is to use the "old" function for the shape index, which leads to okayish results. However, we should look into how to fix it properly... 

```
library(landscapemetrics) # branch rm_na

# (0.25 * perim$value) / sqrt(area$value * 10000)
shape_lsm <- lsm_p_shape(augusta_nlcd)

plot(sort(shape_lsm$value), sort(fragstats_patch_augusta_nlcd$SHAPE))
abline(0, 1)
```

**Comment by mhesselbarth** [comment]
Okay...found it!  The problem is that, I guess due to some numerical issues, in the first conditions m is not equal to 0 for some patches even though it should be (but it is e,g, -9.094947e-13). Interestingly, even `as.integer` did not help.

Since the deviation between `area_patch$value - area_patch$n ^ 2` can not be smaller than 0, I just set all to 0 that are smaller.

Fixed: 18ea976b

**Comment by Nowosad** [comment]
Great work, detective @mhesselbarth !
